### DISCUSSION TITLE
How to compile the Tasmota to enable zigbee2Tasmota

### DISCUSSION THREAD

**Discussion by williamzhou2018** [arendst/Tasmota#15773] [Q&A] [answered]
Hi,
 
   I am trying to build up a zigbee2Tasmota demo in a ESP32-Wrover-E modules with a CC2530EM board.
   I found that there is only unoffical verison of the tasmota-zigbee in ESP32 session. So I am wondering 
1. What 's the difference between offical one and unoffical?
2. how to build up a tasmota-zigbee firmware in ESP32? Anything need to be modified in config files?

tks

**Comment by sfromis** [comment, marked as answer]
The "official" binaries are the ones meant to be supported, relatively stable in content, and the primary suggestions for using. The "unofficial" ones may be experimental, with more variability in content, and higher risk of a binary no longer having what you thought. Therefore they are not as supported, but neither binaries as such are different from what you could build yourself.

**Reply by sfromis** [reply]
You can read about customizing your build here:
https://tasmota.github.io/docs/Compile-your-build/#customize-your-build

Most standard builds are done by preselected build environments, like `env:tasmota-zigbee` in
https://github.com/arendst/Tasmota/blob/development/platformio_tasmota_env.ini

Corresponding builds setups for ESP32 are in
https://github.com/arendst/Tasmota/blob/development/platformio_tasmota_env32.ini
